static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -48,21 +52,37 @@ export default class IntegrationExternalMappingForm extends Component<Props> {
     if (type === 'user') {
       fields.push({
         name: 'userId',
-        type: 'select',
+        type: 'select_async',
         required: true,
         label: tct('Sentry [type]', {type: capitalize(type)}),
         placeholder: t(`Choose your Sentry User`),
-        options,
+        url,
+        onResults: result => {
+          // check if mapping is in result, else add it
+          if (mapping && !result.find(({id}) => id === mapping.userId)) {
```

Comment:
@MeredithAnya @billyvg  We can do this to handle the case of "initial query results does not contain the current mapping, which results in an empty input in the Edit modal". If its not in the initial results, we can just append it. Same pattern for team mapping.

Reply:
@NisanthanNanthakumar this is going to be weird for when there are no search results, because instead of no search results it will just return the initial one. However maybe that's worth it for now cause its going to be buggy either way and this way at least it's populated.

@billyvg thoughts?

Reply:
Hmm I wonder if there's a way to force it to query for the initially selected item?

Reply:
@billyvg this is all still behind a feature flag and I think we want to start getting feedback from users. I think I'd be okay with starting with nisanthan's change and doing a follow up for this for improvements for GA